gui/src-tauri/src/main.rs

```diff
@@ -154,6 +214,58 @@ fn main() {
 			Ok(())
 		})
 		//
-		.run(tauri::generate_context!())
-		.expect("error while running tauri application");
+		.run(tauri::generate_context!());
+	match res {
+		Ok(()) => {}
+		Err(_) => res.expect("error while running tauri application"),
+	}
+}
+
+fn valid_java_paths() -> Vec<(OsString, i32)> {
+	let mut file = Builder::new()
+		.suffix(".class")
+		.tempfile()
+		.expect("Couldn't generate .class file");
+	file.write_all(include_bytes!("JavaVersion.class"))
+		.expect("Couldn't write to .class file");
+	let java_version = file.into_temp_path();
+
+	// Check if main Java is a supported version
+	let main_java = if let Ok(java_home) = std::env::var("JAVA_HOME") {
+		PathBuf::from(java_home).join("bin/java").into_os_string()
+	} else {
+		OsString::from_str("java").unwrap()
+	};
+	if let Some(main_child) = spawn_java(&main_java, java_version.as_os_str())
+		.expect("Couldn't spawn the main Java binary")
+		.wait()
+		.expect("Couldn't execute the main Java binary")
+		.code()
+	{
+		if main_child >= MINIMUM_JAVA_VERSION {
+			return vec![(main_java, main_child)];
+		}
+	}
+
+	// Otherwise check if anything else is a supported version
+	let mut childs = vec![];
+	for java in which_all("java").unwrap() {
+		let res = spawn_java(java.as_os_str(), java_version.as_os_str());
+
+		match res {
+			Ok(child) => childs.push((java.into_os_string(), child)),
+			Err(e) => eprintln!("Error on trying to spawn a Java executable: {}", e),
```

Comment:
I would use `log` here, as we often don't have a mechanism for seeing stderr of tauri. Log will allow us to log to a file in the future if we so choose.

If you want, you can use `pretty_env_logger` as the global logger, for now. Set the default log level to `info`.
You can do that in a followup PR though, I prefer small incremental merges and this PR is already large-ish.
Created #321 to track this.